destination/config.go

```diff
@@ -37,6 +40,22 @@ type Config struct {
 	Key string `json:"key"`
 }
 
+// todo pointer receiver, others are value receivers
+func (c *Config) Validate(context.Context) error {
+	// try parsing the url
+	_, err := pgx.ParseConfig(c.URL)
```

Comment:
why not just inline the err check, and the comment above is redundant, since the function's name is parse ;-)